load/clip.py

```diff
@@ -0,0 +1,338 @@
+
+        for track in self.tracks:
+            if self.frame_on in track.frame_list:
+                track.add_frame(self.frame_on, self.frame_buffer, self.threshold)
+
+    def generate_optical_flow(self):
+        if not self.frame_buffer.has_flow:
+            self.frame_buffer.generate_optical_flow(
+                self.opt_flow, self.config.flow_threshold
+            )
+
+    def start_and_end_time_absolute(self, start_s, end_s):
+        return (
+            self.video_start_time + datetime.timedelta(seconds=start_s),
+            self.video_start_time + datetime.timedelta(seconds=end_s),
+        )
+
+    def get_stats(self):
+        """
+        Returns statistics for this track, including how much it moves, and a score indicating how likely it is
+        that this is a good track.
+        :return: a TrackMovementStatistics record
+        """
+
+        if len(self) <= 1:
+            return track.TrackMovementStatistics()
+
+        # get movement vectors
+        mass_history = [int(bound.mass) for bound in self.bounds_history]
+        variance_history = [bound.pixel_variance for bound in self.bounds_history]
+        mid_x = [bound.mid_x for bound in self.bounds_history]
+        mid_y = [bound.mid_y for bound in self.bounds_history]
+        delta_x = [mid_x[0] - x for x in mid_x]
+        delta_y = [mid_y[0] - y for y in mid_y]
+        vel_x = [cur - prev for cur, prev in zip(mid_x[1:], mid_x[:-1])]
+        vel_y = [cur - prev for cur, prev in zip(mid_y[1:], mid_y[:-1])]
+
+        movement = sum((vx ** 2 + vy ** 2) ** 0.5 for vx, vy in zip(vel_x, vel_y))
+        max_offset = max((dx ** 2 + dy ** 2) ** 0.5 for dx, dy in zip(delta_x, delta_y))
+
+        # the standard deviation is calculated by averaging the per frame variances.
+        # this ends up being slightly different as I'm using /n rather than /(n-1) but that
+        # shouldn't make a big difference as n = width*height*frames which is large.
+        delta_std = float(np.mean(variance_history)) ** 0.5
+
+        movement_points = (movement ** 0.5) + max_offset
+        delta_points = delta_std * 25.0
+        score = min(movement_points, 100) + min(delta_points, 100)
+
+        stats = track.TrackMovementStatistics(
+            movement=float(movement),
+            max_offset=float(max_offset),
+            average_mass=float(np.mean(mass_history)),
+            median_mass=float(np.median(mass_history)),
+            delta_std=float(delta_std),
+            score=float(score),
+        )
+
+        return stats
+
+    def load_tracks(self, metadata, include_filtered_channel):
+        tracks_meta = metadata["tracks"]
+        self.tracks = []
+        # get track data
+        for track_meta in tracks_meta:
+            track = Track(self.get_id())
+            if track.load_track_meta(
+                track_meta, self.frames_per_second, include_filtered_channel
+            ):
+                self.tracks.append(track)
```

Comment:
I'm not a big fan of methods whose output is purely side effects. IMHO, it would be better if this returned the tracks list and the caller did `self.tracks = self.load_tracks(...)`

This makes it easier to see what what's going on at the call site and makes testing easier too (if there were tests!)